src/esphome/binary_sensor/ttp229_lsf_sensor.cpp

```diff
@@ -0,0 +1,96 @@
+#include "esphome/defines.h"
+#ifdef USE_TTP229_LSF
+
+#include "esphome/binary_sensor/ttp229_lsf_sensor.h"
+#include "esphome/log.h"
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace binary_sensor {
+
+static const char *TAG = "binary_sensor.ttp229";
+
+TTP229Channel::TTP229Channel(const std::string &name, int channel_num) : BinarySensor(name) { channel_ = channel_num; }
+
+void TTP229Channel::process(const uint16_t *data, const uint16_t *last_data) {
+  if ((*data & (1 << this->channel_)) && !(*last_data & (1 << this->channel_))) {
+    this->publish_state(true);
+  }
+  if (!(*data & (1 << this->channel_)) && (*last_data & (1 << this->channel_))) {
+    this->publish_state(false);
+  }
+}
+
+TTP229LSFComponent::TTP229LSFComponent(I2CComponent *parent, uint8_t address) : I2CDevice(parent, address) {}
+
+void TTP229LSFComponent::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up ttp229...");
+  if (!this->parent_->raw_request_from(this->address_, 2)) {
+    this->error_code_ = COMMUNICATION_FAILED;
+    this->mark_failed();
+    return;
+  }
+}
+
+void TTP229LSFComponent::dump_config() {
+  ESP_LOGCONFIG(TAG, "ttp229:");
+  LOG_I2C_DEVICE(this);
+  switch (this->error_code_) {
+    case COMMUNICATION_FAILED:
+      ESP_LOGE(TAG, "Communication with ttp229 failed!");
+      break;
+    case WRONG_CHIP_STATE:
+      ESP_LOGE(TAG, "ttp229 has wrong default value for CONFIG2?");
```

Comment:
When can this happen? I don't see other references to wrong_chip_state

Reply:
You marked this as resolved, but as far as I see the comment is still valid.

Reply:
sorry something went wrong when switching branches. I made the change in the wrong branch.
fixed that now.